numpy/core/tests/test_numeric.py

```diff
@@ -1306,6 +1327,202 @@ def test_nonzero_twodim(self):
+            for dtype in types:
+                x = ((2**33)*np.random.randn(10, 10, 3)).astype(dtype)
+                np.random.shuffle(idxs)
+                num_zeros = np.random.randint(0, 301)
+                x.flat[idxs[:num_zeros]] = 0
+                np.random.shuffle(dim_permutation)
+                x = np.transpose(x, dim_permutation)
+                idxs_0, idxs_1, idxs_2 = np.nonzero(x)
+                assert_equal(np.count_nonzero(x), len(idxs_0))
+                nzs = 0
+                for i,j,k in zip(idxs_0, idxs_1, idxs_2):
+                    if x[i,j,k] == 0:
+                        nzs += 1
+                assert_equal(nzs, 0)
+
+        # Repeat the above for fortran contiguous arrays
+        for _ in range(iters):
+            for dtype in types:
+                x = ((2**33)*np.random.randn(10, 10, 3)).astype(dtype)
+                x = np.asfortranarray(x)
+                np.random.shuffle(idxs)
+                num_zeros = np.random.randint(0, 301)
+                x.flat[idxs[:num_zeros]] = 0
+                idxs_0, idxs_1, idxs_2 = np.nonzero(x)
+                assert_equal(np.count_nonzero(x), len(idxs_0))
+                nzs = 0
+                for i,j,k in zip(idxs_0, idxs_1, idxs_2):
+                    if x[i,j,k] == 0:
+                        nzs += 1
+                assert_equal(nzs, 0)
+
+
+        for _ in range(iters): # This time randomly permute dimensions
+            for dtype in types:
+                x = ((2**33)*np.random.randn(10, 10, 3)).astype(dtype)
+                x = np.asfortranarray(x)
+                np.random.shuffle(idxs)
+                num_zeros = np.random.randint(0, 301)
+                x.flat[idxs[:num_zeros]] = 0
+                np.random.shuffle(dim_permutation)
+                x = np.transpose(x, dim_permutation)
+                idxs_0, idxs_1, idxs_2 = np.nonzero(x)
+                assert_equal(np.count_nonzero(x), len(idxs_0))
+                nzs = 0
+                for i,j,k in zip(idxs_0, idxs_1, idxs_2):
+                    if x[i,j,k] == 0:
+                        nzs += 1
+                assert_equal(nzs, 0)
+
+
+        for i in range(iters): # Check for slices
+            x = ((2**33)*np.random.randn(10, 10)).astype(np.int32)
+            x.flat[[2,6,9,15,21]] = 0
+            x_view = x[:5,:5]
+
+            idxs_0, idxs_1 = np.nonzero(x_view)
+            assert_equal(np.count_nonzero(x_view), len(idxs_0))
+            nzs = 0
+            for i,j in zip(idxs_0, idxs_1):
+                if x_view[i,j] == 0:
+                    nzs += 1
+            assert_equal(nzs, 0)
+
+            y = np.asfortranarray(x)
+            y_view = y[:5,:5]
+
+            idxs_0, idxs_1 = np.nonzero(y_view)
+            assert_equal(np.count_nonzero(y_view), len(idxs_0))
+            nzs = 0
+            for i,j in zip(idxs_0, idxs_1):
+                if y_view[i,j] == 0:
+                    nzs += 1
+            assert_equal(nzs, 0)
```

Comment:
And one more comment that got losts. These tests seem very repitetive, can you try to use `pytest.mark.parametrize` or create a helper function like `check_nonzero_result`?